WMF Framework/Event Platform Client/EPCNetworkManager.swift

```diff
@@ -0,0 +1,140 @@
+
+import Foundation
+
+class EPCNetworkManager: EPCNetworkManaging {
+    
+    private let storageManager: EPCStorageManaging
+    private let session: Session
+    private let operationQueue: OperationQueue
+    
+    init(storageManager: EPCStorageManaging, session: Session = Session.shared) {
+        self.storageManager = storageManager
+        self.session = session
+        
+        operationQueue = OperationQueue()
+        operationQueue.maxConcurrentOperationCount = 1
+    }
+    
+    func httpPost(url: URL, body: NSDictionary) {
+        storageManager.createAndSavePost(with: url, body: body)
+    }
+    
+    func httpDownload(url: URL, completion: @escaping (Data?) -> Void) {
+        httpDownload(url: url, attempt: 0, maxAttempts: 5, attemptDelay: 2, completion: completion)
+    }
+    
+    private func httpDownload(url: URL, attempt: Int, maxAttempts: Int, attemptDelay: TimeInterval, completion: @escaping (Data?) -> Void) {
+        
+        guard attempt < maxAttempts else {
+            completion(nil)
+            return
+        }
+        
+        let task = session.dataTask(with: url) { (data, response, error) in
+            
+            let failureBlock = {
+                dispatchOnMainQueueAfterDelayInSeconds(attemptDelay) {
+                    self.httpDownload(url: url, attempt: attempt + 1, maxAttempts: maxAttempts, attemptDelay: attemptDelay, completion: completion)
+                }
+            }
```

Comment:
@tonisevener This is clever!! I like this a lot! I think I improved it slightly in an upcoming commit by getting rid of `attempt`, switching to `maxAttempts: maxAttempts - 1`, and checking for `maxAttempts > 0`